database/src/backend/heed/database.rs

```diff
@@ -200,59 +200,63 @@ impl<T: Table> DatabaseRw<T> for HeedTableRw<'_, '_, T> {
 
     #[inline]
     fn delete(&mut self, key: &T::Key) -> Result<(), RuntimeError> {
-        self.db.delete(&mut self.tx_rw.borrow_mut(), key)?;
-        Ok(())
+        match self.db.delete(&mut self.tx_rw.borrow_mut(), key) {
+            Ok(true) => Ok(()),
+            Ok(false) => Err(RuntimeError::KeyNotFound),
+            Err(e) => Err(e.into()),
+        }
```

Comment:
I don't see how we could fall in that case. I agree the outcome is the same, but its worth logging it, as it will be an indicator that somehow we tried to.
```rust
if !self.db.delete(&mut self.tx_rw.borrow_mut(), key)? {
    warn!("Tried to delete non existent key") // replace with println
}
```

Reply:
My bad actually, this would have caused behavior differences between our current `redb` impl.

Changed back to returning `Ok(())` if key doesn't exist.